### PR TITLE
fix(sqlite): always use "main" as db name

### PR DESCRIPTION
**Pull request by Weakky**
Fixes https://github.com/prisma/prisma-engines/issues/1481

Follow-up PR to https://github.com/prisma/quaint/pull/252. Check it out for more information.

### CODE REVIEW

File: query-engine/connectors/sql-query-connector/src/database/sqlite.rs
Comment (Weakky) [inline query-engine/connectors/sql-query-connector/src/database/sqlite.rs:45]: Isn't there a more rust idiomatic way to express that?

File: query-engine/connectors/sql-query-connector/src/database/sqlite.rs
Comment (Weakky) [inline query-engine/connectors/sql-query-connector/src/database/sqlite.rs:61]: We no longer add `db_name` to the connection string. Quaint no longer supports it anyway since https://github.com/prisma/quaint/pull/252. Instead, we use the default db name called `main`.

File: query-engine/connectors/sql-query-connector/src/database/sqlite.rs
Comment (dpetrick) [reply query-engine/connectors/sql-query-connector/src/database/sqlite.rs:45]: You could use the Url crate (should be loaded in the connectors already) instead and use one of their methods to get the path (or whatever exactly you need here).

**Review by dpetrick — APPROVED** [review]

**Comment by pimeys** [comment]
This might be a fix for issue https://github.com/prisma/prisma/issues/2955

File: query-engine/connectors/sql-query-connector/src/database/sqlite.rs
Comment (pimeys) [reply query-engine/connectors/sql-query-connector/src/database/sqlite.rs:45]: The crate doesn't work here because it kind of messes up the file paths. We used to use that, but had too many issues and go with `PathBuf` nowadays...

Comment (pimeys) [reply query-engine/connectors/sql-query-connector/src/database/sqlite.rs:45]: This would be a bit shorter:
```rust
if url.map(Path::new).and_then(|url| url.file_stem()).is_nume() {
    return Err(invalid_file_path_error(&file_path, &connection_info));
}
```

**Comment by pimeys** [comment]
I didn't see `Cargo.lock` being updated, but please do `cargo update -p quaint` to this branch before merging.

File: query-engine/connectors/sql-query-connector/src/database/sqlite.rs
Comment (Weakky) [reply query-engine/connectors/sql-query-connector/src/database/sqlite.rs:45]: I was initially mostly referring to the `if X || Y { return Err() }` - Maybe the question is more: Is there a more functional way of returning the error.

**Comment by Weakky** [comment]
> I didn't see `Cargo.lock` being updated, but please do `cargo update -p quaint` to this branch before merging.

Yeah, I was waiting for quaint to be released. Done now, will wait for the tests to pass and merge!

**Comment by Weakky** [comment]
PR was rebased on https://github.com/prisma/prisma-engines/pull/1652 to enable merging with green lights. Closing this now.

**Pull request closed without merging**
